### PR TITLE
Use medical record disability fields instead of listing all traits

### PR DESCRIPTION
**Pull request by pgmzeta**
[QOL][STATION-SYTEMS][MEDICAL]

## About the PR 
Instead of listing all character traits in medical records, add major and minor disabilities based on the following traits:
* Major: Deaf, Blind, Stumped, Puritan
* Minor: Short-sighted, Hemophilia, Frail Constitution, Hyperallergic

These specific traits were included because they are medically relevant and likely to be recorded in medical records. Disability names use medical-themed verbiage instead of the trait name verbatim.

## Why's this needed? 

Make it clearer to doctors what the impact of traits to their job duties in an unobtrusive and in-universe way.

"Don't clone a puritan!!!" How would you know that unless you *already* knew what the Puritan trait was/meant?

## Screenshot
An in-game screenshot showing a medical record with a large number of disabilities via both PDA and MedTrak.
image

## Changelog 

```changelog
(u)glowbold
(*)Medical records now list relevant traits as major/minor disabilities instead of a full list.
```

### CODE REVIEW

File: code/datums/datacore.dm [hunk 1]
```diff
@@ -88,21 +88,72 @@
...
 			if (istype(T, /datum/trait/random_allergy))
 				var/datum/trait/random_allergy/AT = T
 				if (M["alg"] == "None") //is it in its default state?
 					M["alg"] = reagent_id_to_name(AT.allergen)
 					M["alg_d"] = "Allergy information imported from CentCom database."
 				else
 					M["alg"] += ", [reagent_id_to_name(AT.allergen)]"
+				continue
+
+			if(istype(T, /datum/trait/deaf))
```
Comment (TobleroneSwordfish) [inline code/datums/datacore.dm:114]: Don't do this with a big if tree, put the data on the trait instead

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Cool idea, makes sense

**Review by TobleroneSwordfish — APPROVED** [review]

File: code/datums/datacore.dm [hunk 2]
```diff
@@ -87,21 +87,38 @@
...
+
+			switch(T.disability_type)
+				if (TRAIT_DISABILITY_MAJOR)
+					majorDisabilities.Add(T.disability_name)
+					majorDisabilityDesc.Add(T.disability_desc)
+				if (TRAIT_DISABILITY_MINOR)
+					minorDisabilities.Add(T.disability_name)
+					minorDisabilityDesc.Add(T.disability_desc)
+
+	if(minorDisabilities.len)
```
Comment (ZeWaka) [inline code/datums/datacore.dm:116]:
```suggestion
	if(length(minorDisabilities))
```

File: code/datums/datacore.dm [hunk 3]
```diff
@@ -87,21 +87,38 @@
...
+		M["mi_dis"] = jointext(minorDisabilities, ", ")
+		M["mi_dis_d"] = jointext(minorDisabilityDesc, ". ")
+	if(majorDisabilities.len)
```
Comment (ZeWaka) [inline code/datums/datacore.dm:119]:
```suggestion
	if(length(majorDisabilities))
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**